metaflow/plugins/kfp/kfp.py

```diff
@@ -779,6 +792,40 @@ def _update_step_op_func_signature(
         func.__signature__ = new_sig
         return func
 
+    def _create_s3_sensor_op(
+        self, s3_sensor_deco: FlowDecorator, flow_parameters_json: str
+    ) -> ContainerOp:
+        path = s3_sensor_deco.path
+        timeout_seconds = s3_sensor_deco.timeout_seconds
+        polling_interval_seconds = s3_sensor_deco.polling_interval_seconds
+        path_formatter = s3_sensor_deco.path_formatter
+
+        # see https://github.com/kubeflow/pipelines/pull/1946/files
+        # KFP does not support the serialization of Python functions directly. The KFP team took
+        # the approach of using base64 encoding + pickle. Pickle didn't quite work out
+        # in this case because pickling a function directly stores references to the function's path,
+        # which couldn't be resolved when the path_formatter function was unpickled within the running
+        # container. Instead, we took the approach of marshalling just the code of the path_formatter
+        # function, and reconstructing the function within the kf_s3_sensor.py code.
+        path_formatter_code_encoded = base64.b64encode(
+            marshal.dumps(path_formatter.__code__)
+        ).decode("ascii")
+
+        s3_sensor_op = func_to_container_op(
+            wait_for_s3_path,
+            base_image="hsezhiyan/s3_sensor:1.0",
```

Comment:
Do you need your custom docker image?  Let's not have a loose dependency on `hsezhiyan` because you may become a millionaire and not work with us anymore.

The image only needs Python and boto3 installed. Why not use https://github.com/zillow/metaflow/blob/ecac3fb5cdaa9074c6bc621d31769f84e4d05430/metaflow/plugins/kfp/kfp_constants.py#L4-L4.

Can you please create a backlog task to have this Docker file in this repo in `kfp` dir?

Reply:
Here's the backlog ticket: https://zbrt.atl.zillow.net/browse/AIP-4571